core/server/config/url.js

```diff
@@ -205,10 +205,12 @@ function urlFor(context, data, absolute) {
             hostname = baseUrl.split('//')[1] + ghostConfig.paths.subdir;
             if (urlPath.indexOf(hostname) > -1
                 && urlPath.indexOf('.' + hostname) === -1
-                && urlPath.indexOf('mailto:') !== 0) {
+                && urlPath.indexOf('mailto:') !== 0
+                && !urlPath.split(hostname)[0].match(/\./)) {
```

Comment:
with this new condition you don't need to keep `urlPath.indexOf('.' + hostname) === -1`, it's the same result in the case of sub domain.

you can also remove `urlPath.indexOf('mailto:') !== 0` if you change your match regex to `!urlPath.split(hostname)[0].match(/\.|mailto/)`

suggestion:
I think it's not obvious to understand why we don't make the link relative in this case. can you add a comment to describe what happens if we match `.`